File: python/hamming/hamming.py
def distance(strand_a, strand_b):
    if (len(strand_a) != len(strand_b)):
        raise ValueError('Unequal length dna strands')

    hamming = 0
    i = 0
    while i < len(strand_a):
        if (strand_a[i] != strand_b[i]):
            hamming += 1
        i += 1

    return hamming

File: cpp/hamming/hamming.cpp
#include "hamming.h"
#include <string>
#include <iostream>
#include <stdexcept>

int hamming::compute(std::string a, std::string b) {
    if (a.length() != b.length()) {
        throw std::domain_error("Unequal length dna strings.");
    }
    
    int hamming_distance = 0;
    int size = a.length();

    for (int i = 0; i < size; i++) {
        if (a[i] != b[i]) {
            hamming_distance++;
        }
    }

    return hamming_distance;
}